ValidParenthesis.java

```diff
@@ -0,0 +1,37 @@
+/**
+Leet Code Submitted : YES
+Time Complexity : O(N)
+Space Complexity : O(N/2)
```

Comment:
Space is O(n) in worst case. Imagine whole string has only opening brackets ....